probe-rs/src/vendor/st/sequences/stm32_armv7.rs

```diff
@@ -21,12 +21,16 @@ use crate::architecture::arm::{
 
 /// Marker structure for most ARMv7 STM32 devices.
 #[derive(Debug)]
-pub struct Stm32Armv7 {}
+pub struct Stm32Armv7 {
+    saved_cr_value: Mutex<Option<u32>>,
```

Comment:
I dont think the Mutex is necessary as the sequences are held on the session which should then go into the mutex. Not internally.

Reply:
`Stm32Armv7` implements `ArmDebugSequence` which is `where Self: Sync`. So I think I need something thread-safe here. If not `Mutex`, what should I use instead?

Reply:
You can just have the Option<u32> which you then can get a reference to and the sequence should be passed mutably so you can access it if you hold an &mut to the thing :)

Reply:
Also, [Option](https://doc.rust-lang.org/std/option/enum.Option.html#impl-Sync-for-Option%3CT%3E) is `Sync` as long as `T:Sync` and `u32` is also Sync.

Reply:
`debug_device_unlock` takes `&self`, not `&mut self`. And it cannot be (easily) changed to take `&mut` because its owner holds it in an `Arc`, and that `Arc` is always cloned, see https://github.com/probe-rs/probe-rs/blob/956a2bd745d8d499b3b0a3b6a562d533c8e624c7/probe-rs/src/session.rs#L185

Reply:
Ah, I disregarded that fact. I would prefer no Arc 🤔 Maybe we can reevaluate in the future.